packages/config-loader/src/loader.ts

```diff
@@ -107,6 +107,7 @@ export async function loadConfig(
     remote: options.remote && {
       reloadInterval: { seconds: options.remote.reloadIntervalSeconds },
     },
+    watch: options.watch,
```

Comment:
Ah so this is the point where we convert the legacy watch option to the new one and it is disabled by default, so this should be `watch: Boolean(options.watch)`, sorry if I confused you here before